node/api/renter.go

```diff
@@ -373,6 +379,15 @@ func (api *API) renterRenameHandler(w http.ResponseWriter, req *http.Request, ps
 	WriteSuccess(w)
 }
 
+// renterFileHandler handles the API call to return specific file.
+func (api *API) renterFileHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	file, err := api.renter.File(strings.TrimPrefix(ps.ByName("siapath"), "/"))
+	WriteJSON(w, RenterFile{
+		File:  file,
+		Error: err,
+	})
+}
```

Comment:
instead of including the error in the JSON response, this endpoint should probably signal the error using a non-200 status code, e.g.
```go
file, err := api.renter.File(strings.TrimPrefix(ps.ByName("siapath"), "/"))
if err != nil {
    WriteError(w, Error{err.Error()}, http.StatusBadRequest)
    return
}
WriteJSON(w, file)
```

Reply:
@lukechampine will passing file into WriteJSON like like that negate the need for a RenterFile struct?

Also would that then make the resulting JSON be like so?
```
{
    {
      "siapath":        "foo/bar.txt",
      "localpath":      "/home/foo/bar.txt",
      "filesize":       8192, // bytes
      "available":      true,
      "renewing":       true,
      "redundancy":     5,
      "bytesuploaded":  209715200, // total bytes uploaded
      "uploadprogress": 100, // percent
      "expiration":     60000
    }
}
```